lib/pure/collections/sets.nim

```diff
@@ -907,7 +909,52 @@ iterator pairs*[A](s: OrderedSet[A]): tuple[a: int, b: A] =
   forAllOrderedPairs:
     yield (idx, s.data[h].key)
 
-
+proc fromJsonHook*[A, B](s: var SomeSet[A], jsonNode: B) =
+  ## Enables `fromJson` for `HashSet` and `OrderedSet` types.
+  ## 
+  ## See also:
+  ## * `toJsonHook proc<#toJsonHook,SomeSet[A]>`_
+  runnableExamples:
```

Comment:
please also update tests/stdlib/tjsonutils.nim (with HashSet, OrderedSet, and if not already present, Table and OrderedTable)

Reply:
Done.